File: DMOJ/dmoj.ca-downloaded-submissions/ccc15j3.cpp
#include <iostream> 
#include <string> 

using namespace std;

string alphabet = "abcdefghijklmnopqrstuvwxyz"; 
string vowels = "aeiou";

char closestVowels(char letter){ 
    int smallest = 1000000; 
    char closestVowel; 
    int current = 0; 
    for (int i = 0; i < 5; i ++){
        current = abs((int)letter - (int)vowels[i]);
        if (current < smallest){
            smallest = current; 
            closestVowel = vowels[i];
        }
    }
    return closestVowel;    
}

char nextConsonant(char letter){
    if (letter == 'z'){
        return 'z'; 
    }else if (alphabet[(int)letter - 97 + 1] == 'a' || alphabet[(int)letter - 97 + 1] == 'e' || alphabet[(int)letter - 97 + 1] == 'i' || alphabet[(int)letter - 97 + 1] == 'o' || alphabet[(int)letter - 97 + 1] == 'u'){
        return alphabet[(int)letter - 97 + 2];
    }else {
        return alphabet[(int)letter - 97 + 1];
    }
}

int main(){
    string initial, final; 
    cin >> initial; 
    for (int i = 0; i < initial.length(); i ++){
        if (initial[i] != 'a' && initial[i] != 'e' && initial[i] != 'i' && initial[i] != 'o' && initial[i] != 'u'){
            final += initial[i]; 
            final += closestVowels(initial[i]); 
            final += nextConsonant(initial[i]);
        }else{ 
            final += initial[i];
        }
    }

    cout << final; 
    return 0; 
}

File: DMOJ/dmoj.ca-downloaded-submissions/ccc15j3.py
vowels = ["a", "e", "i", "o", "u"]

closest_vowels = {
    "b" : "a",
    "c" : "a",
    "d" : "e",
    "f" : "e",
    "g" : "e",
    "h" : "i",
    "j" : "i",
    "k" : "i",
    "l" : "i",
    "m" : "o",
    "n" : "o",
    "p" : "o",
    "q" : "o",
    "r" : "o",
    "s" : "u",
    "t" : "u",
    "v" : "u",
    "w" : "u",
    "x" : "u",
    "y" : "u",
    "z" : "u"
}

next_consonant = {
    "b" : "c",
    "c" : "d",
    "d" : "f",
    "f" : "g",
    "g" : "h",
    "h" : "j",
    "j" : "k",
    "k" : "l",
    "l" : "m",
    "m" : "n",
    "n" : "p",
    "p" : "q",
    "q" : "r",
    "r" : "s",
    "s" : "t",
    "t" : "v",
    "v" : "w",
    "w" : "x",
    "x" : "y",
    "y" : "z",
    "z" : "z",
}

inputy = input()
word = list(inputy)

wodry = []
for element in word:
    if element not in vowels: 
        element = element + closest_vowels[element] + next_consonant[element]
        wodry.append(element)
    else:
        wodry.append(element)
    
finstring = ""
for element in wodry:
    finstring += element

print(finstring)